### PR TITLE
Slight correction in temperature explanation 

### PR DESCRIPTION
**Pull request by nvassilyev**
For `0.0<t<1.0`, this actually pushes the probability distribution closer together making it more "sharp" or "peaked". Hence why when t approaches 0, the probability distribution becomes more and more squished towards the most likely tokens and at one point collapsing to the single most likely token (which is why it becomes greedy decoding). This huggingface blog has a very nice overview of the decoding methods:
https://huggingface.co/blog/how-to-generate

### CODE REVIEW

**Comment by stas00** [comment]
Thank you for suggestion to improve the temp section,

I find your shift is more difficult to easily understand.

My original text is trying to describe this:
temp-probability
so you can see when values are pushed together or apart as the original text says.

I think think we just need to `s/logit probabilities/probabilities` in my original text as it's sort of confusing.

Perhaps including that diagram will make things crystal clear?

**Comment by nvassilyev** [comment]
Ah yes I see the point you were trying to make. I think I misinterpreted your original explanation as I associated "the logit probabilities are pushed further apart" visually with a probability distribution being stretched horizontally and "the logit probabilities are pushed closer together" with the probability distribution being squished towards a more unimodal shape. 

I proposed a different a description but if this is still unclear, I agree we can revert to the description you already had as the diagram will prevent a similar misunderstanding.

**Comment by stas00** [comment]
Yeah, I don't think the addition of "probability distribution" notes helps the reader to quickly understand the outcome. I'd remove them and let the picture speak for itself, where the original "closer" and "further apart" describe the picture.

**Comment by nvassilyev** [comment]
Reverted it back to what it was before. What are your thoughts on the followup explanation of temperature or do you prefer the original passage as well?

> The softmax operation turns logit differences into probability ratios - when we divide by t<1.0, we make these differences larger, causing more extreme probability ratios and a more peaked distribution. When we divide by t>1.0, we make these differences smaller, causing more similar probability ratios and a more uniform distribution. At t=0, this effectively makes the highest logit infinitely larger than the others (though division by zero is avoided in practice).

**Comment by stas00** [comment]
That additional explanation is great, Nikita.

We can merge this if you feel it's good now from your side.

**Comment by stas00** [comment]
It's awesome to work with you again, Nikita!

Thank you for your contribution!
